src/NexusMods.Library/DownloadsService.cs

```diff
@@ -0,0 +1,330 @@
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Reactive.Disposables;
+using DynamicData;
+using DynamicData.Kernel;
+using NexusMods.Abstractions.Downloads;
+using NexusMods.Abstractions.NexusWebApi.Types.V2;
+using NexusMods.Abstractions.Jobs;
+using NexusMods.Abstractions.Library.Models;
+using NexusMods.MnemonicDB.Abstractions;
+using NexusMods.Paths;
+using NexusMods.Networking.HttpDownloader;
+using NexusMods.Networking.NexusWebApi;
+using NexusMods.Abstractions.NexusModsLibrary;
+using ReactiveUI;
+
+namespace NexusMods.Library;
+
+/// <summary>
+/// Implementation of <see cref="IDownloadsService"/>.
+/// </summary>
+public sealed class DownloadsService : IDownloadsService, IDisposable
+{
+    // TODO: Localize this string
+    private const string UnknownDownloadName = "Unknown Download";
+    
+    private readonly IJobMonitor _jobMonitor;
+    private readonly IConnection _connection;
+    private readonly CompositeDisposable _disposables = new();
+    
+    // Mapping from JobId to DownloadId to maintain consistency
+    private readonly ConcurrentDictionary<JobId, DownloadId> _jobIdToDownloadId = new();
```

Comment:
As discussed, we should make `DownloadId` inherit the value from `JobId` directly, to avoid having to maintain a dictionary and still get efficient lookups. 

This should reduce code complexity down a bit.